version: '3.8'

services:
  logstash-app:
    image: docker.elastic.co/logstash/logstash:7.10.2
    container_name: logstash-app
    environment:
      # LS_JAVA_OPTS: -Xms512m -Xmx512m
      # LOGSTASH_INTERNAL_PASSWORD: ${LOGSTASH_INTERNAL_PASSWORD:-}
      MONITORING_ENABLED: true
    volumes:
      - ./app/logstash/config/logstash.yml:/usr/share/logstash/config/logstash.yml:ro,Z
      # - ./logstash/pipeline:/usr/share/logstash/pipeline:ro,Z
      - ./app/logstash/config/logstash-test.conf:/usr/share/logstash/config/logstash-test.conf:ro,Z
      - ./logs/elasticsearch:/var/log/elasticsearch
    networks:
      - elastic  
    # no : 재시작 하지 않음.
    # always : exit code와 상관없이 항상 재시작.
    # on-failure : exit code가 0일 아닐때만 재시작
    # unless-stopped : 컨테이너 상태를 수동으로 stop 하기전까지 재시작.
    restart: unless-stopped

networks:
  elastic:
    driver: bridge